### PR TITLE
CBG-2739 make some ISGR tests pass with non-default collections

### PR DESCRIPTION
**Pull request by torcolvin**
Not all tests pass, and many tests are marked as not working, but I didn't want to make this code review even bigger.

Definitely review with whitespace off.

## Pre-review checklist
- [x] Removed debug logging (`fmt.Print`, `log.Print`, ...)
- [x] Logging sensitive data? Make sure it's tagged (e.g. `base.UD(docID)`, `base.MD(dbName)`)
- [x] Updated relevant information in the API specifications (such as endpoint descriptions, schemas, ...) in `docs/api`

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGateway-Integration/build?delay=0sec)
- [x] `GSI=true,xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/1584/

### CODE REVIEW

File: rest/replicatortest/replicator_collection_test.go [hunk 1]
```diff
@@ -45,6 +45,10 @@ func TestActiveReplicatorMultiCollection(t *testing.T) {
 		numDocsPerCollection = 3
 	)
 
+	base.RequireNumTestBuckets(t, 2)
+	base.RequireNumTestDataStores(t, numCollections)
+	base.TestRequiresCollections(t)
```
Comment (bbrks) [inline rest/replicatortest/replicator_collection_test.go]: 2 of these are already at the top of the test

File: rest/utilities_testing_resttester.go [hunk 2]
```diff
@@ -204,42 +208,48 @@ func (rt *RestTester) GetReplicationStatuses(queryString string) (statuses []db.
 //	  - user 'alice' created with star channel access
 //	  - http server wrapping the public API, remoteDBURLString targets the rt2 database as user alice (e.g. http://alice:pass@host/db)
 //	returned teardown function closes activeRT, passiveRT and the http server, should be invoked with defer
-func SetupSGRPeers(t *testing.T) (activeRT *RestTester, passiveRT *RestTester, remoteDBURLString string, teardown func()) {
+func SetupSGRPeers(t *testing.T, useMultipleCollections bool) (activeRT *RestTester, passiveRT *RestTester, remoteDBURLString string, teardown func()) {
```
Comment (bbrks) [inline rest/utilities_testing_resttester.go:211]: Is this acting more like a `useDefaultCollection` bool (if the condition on 222 is reversed) ?

**Review by bbrks — CHANGES_REQUESTED** [review]
Just a couple of things I spotted

File: rest/utilities_testing_resttester.go (hunk 2 above)
Comment (torcolvin) [reply rest/utilities_testing_resttester.go:211]: Yeah. I could change this if it bothers you, I'd just flip the booleans.

**Review by bbrks — APPROVED** [review]

**Pull request merged**